VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -161,38 +166,38 @@ public AuthResponse registerUsingFacebook(@NonNull Bundle parameters, @NonNull S
     }
 
     @NonNull
-    public ResetPasswordResponse resetPassword(@NonNull String email) throws HttpException {
-        return loginService.resetPassword(email);
-    }
-
-    @NonNull
     private void register(Bundle parameters) throws Exception {
-        try {
-            final Map<String, String> parameterMap = new HashMap<>();
-            for (String key : parameters.keySet()) {
-                parameterMap.put(key, parameters.getString(key));
-            }
-            loginService.register(parameterMap);
-        } catch (HttpResponseStatusException e) {
-            if ((e.getStatusCode() == 400 || e.getStatusCode() == 409) && !android.text.TextUtils.isEmpty(e.getBody())) {
+        final Map<String, String> parameterMap = new HashMap<>();
+        for (String key : parameters.keySet()) {
+            parameterMap.put(key, parameters.getString(key));
+        }
+        Response<ResponseBody> response = loginService.register(parameterMap).execute();
+        if (!response.isSuccessful()) {
+            final int errorCode = response.code();
+            final String errorBody = response.errorBody().string();
+            if ((errorCode == 400 || errorCode == 409) && !android.text.TextUtils.isEmpty(errorBody)) {
```

Comment:
400 and 409 error codes are available as constants from the `HttpURLConnection` class, so they could be replaced with those here. However, we need to have a comprehensive list of constants for HTTP status codes, which I've implemented at #748.